### PR TITLE
#162414172 create article tags

### PR DESCRIPTION
**Pull request by iMichaelOwolabi**
#### What does this PR do?
Implement the create tag controller with validation and route
#### Description of Task to be completed?
check and filter article tags from user input via the params
The application allows users to be able to create tags alongside their articles and this PR filters the already created tags as the users are typing.
#### How should this be manually tested?
* copy and paste this URL on postman localhost:3000/api/v1/tags/
* add any string at the end of the URL and make a GET request
* You will get a response whether or not there is a tag that matches the input string on the URL
#### Any background context you want to provide?
The application allows users to be able to create tags alongside their articles and this PR filters the already created tags as the users are typing.
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162414172
#### Screenshots (if appropriate)
screenshot 2019-01-25 at 5 22 05 pm
screenshot 2019-01-25 at 5 25 14 pm

#### Questions:

### CODE REVIEW

**Review by ibraheemkabir — DISMISSED** [review]

File: test/emailVerification/sendVerificationMail.test.js [hunk 1]
```diff
@@ -5,7 +5,12 @@ const sinonChai = require('sinon-chai');
...
-const { FROM, SUBJECT, SERVER_ERROR_CODE } = require('../../constants');
+const {
+  FROM,
+  SUBJECT,
+  SERVER_ERROR_CODE,
+  RESOURCE_CREATED_CODE
+} = require('../../constants');
 
 chai.use(sinonChai);
 const validToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1dWlkIjoiNDE1ZTJlZmEtMTllNy0xMWU5LWFiMTQtZDY2M2JkODczZDkzIiwiaWF0IjoxNTQ3ODIxMjk2LCJleHAiOjE1NDc5MDc2OTZ9.DVwT7E9aU92ByHEw7GNn8URI-iZAR2VCLIY47dilOec';
```
Comment (iAfolayan) [inline test/emailVerification/sendVerificationMail.test.js]: Kindly move this validToken to constant file.

**Review by iAfolayan — DISMISSED** [review]

File: test/emailVerification/sendVerificationMail.test.js (hunk 1 above)
Comment (iMichaelOwolabi) [reply test/emailVerification/sendVerificationMail.test.js]: fixed

File: controllers/filterTagsController.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  TAGS_NOT_FOUND,
+} = require('../constants');
+
+const filterTags = async (req, res) => {
+  const { tag } = req.params;
+  const foundTags = await TagModelManager.getTag(tag);
+  const matchingTags = [];
+  if (foundTags.length === 0) {
+    return failureResponse(res, TAGS_NOT_FOUND);
+  }
```
Comment (codrex) [inline controllers/filterTagsController.js:14]:
```js
  if (foundTags.length === 0) {
    return failureResponse(res, TAGS_NOT_FOUND);
  }
```
It is best to return the empty array return by the database to the user when records are not found.

File: controllers/filterTagsController.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  foundTags.forEach((tags) => {
+    matchingTags.push(tags.dataValues.tagName);
+  });
```
Comment (codrex) [inline controllers/filterTagsController.js]: `foundTags.map` is more suited for this type of operation

File: controllers/filterTagsController.js (hunk 2 above)
Comment (iMichaelOwolabi) [reply controllers/filterTagsController.js:14]: Ok. will effect the change

File: controllers/filterTagsController.js (hunk 3 above)
Comment (iMichaelOwolabi) [reply controllers/filterTagsController.js]: Noted and fixed. Thanks for the heads up

**Review by ibraheemkabir — APPROVED** [review]

**Review by tosinibrahim96 — APPROVED** [review]

**Review by codrex — APPROVED** [review]

**Pull request merged**